### ISSUE TITLE
'pickBy' incompatible with 3.x 'pick'

### ISSUE THREAD

**Issue by Oceanswave** [lodash/lodash#1848]
The predicate in 3.x previously was passed the property key as the 2nd argument. Without knowing the property key, it's no longer possible to dynamically pick based on value+key.

For instance, in 3.x I want an object that omits all properties that are prefixed with _, in 3.x this was:

```
_.pick(foo, function (value, key) {
                    return key.charAt(0) !== "_";
});
```

in 4.0, key is no longer passed with pickBy...

```
_.pickBy(foo, function (value, key) {
                    return key.charAt(0) !== "_";
});

//foo always equals {};
```

**Comment by Oceanswave** [comment]
Um. I think I got it -- the replacement seems to be .mapKeys

**Comment by d10** [comment]
This has been fixed with 3eaa7b3, after the discussion at #1765.
